### DISCUSSION TITLE
How to use picsimlab in CI/CD?

### DISCUSSION THREAD

**Discussion by kassane** [lcgamboa/picsimlab#121] [Q&A] [answered]
It is possible to run without a graphical interface.
However, I had not yet seen GH action using picsimlab as an alternative to wokwi on some devices.

**Comment by lcgamboa** [comment]
I believe it is possible, but I don´t have time to implement it yet. 
The simplest way would be to write a client program using the PICSimLab [rcontrol](https://lcgamboa.github.io/picsimlab_docs/stable/RemoteControlInterface.html) protocol, like in the [PICSimLab tests.](https://github.com/lcgamboa/picsimlab/tree/master/tests) The tests are written in C++ but could be written in Python or any other language that supports TCP/IP sockets.
I have not yet used or know of any projects that use Wokwi CI. 
I currently use the PICSimLab rcontrol interface to automate the correction of exercises in the microcontroller laboratory course.

**Reply by lcgamboa** [reply, marked as answer]
Take a look at https://github.com/lcgamboa/picsimlab_python_rcontrol/tree/main and in the new PICSimLab code project wizard that generates a code template with automated test.
